source/common/event/file_event_impl.cc

```diff
@@ -130,6 +151,14 @@ void FileEventImpl::mergeInjectedEventsAndRunCb(uint32_t events) {
     injected_activation_events_ = 0;
     activation_cb_->cancel();
   }
+
+  std::array<uint32_t, 2> event_types{Event::FileReadyType::Write, Event::FileReadyType::Read};
```

Comment:
I agree with the suggested use of oneshot since it does exactly this for you

Reply:
I think since the implementation does not change by much I will add it as a TODO to improve on the existing implementation with oneshot.